crates/astria-sequencer/src/authority/state_ext.rs

```diff
@@ -50,31 +54,33 @@ impl ValidatorSet {
     }
 
     #[cfg(test)]
-    pub(crate) fn get(
-        &self,
-        pub_key: &tendermint::public_key::PublicKey,
-    ) -> Option<&tendermint::validator::Update> {
-        self.0.get(&pub_key.to_hex())
+    pub(crate) fn get(&self, address: &account::Id) -> Option<&validator::Update> {
+        self.0.get(address)
+    }
+
+    pub(crate) fn push_update(&mut self, update: validator::Update) {
+        let address = tendermint::account::Id::from(update.pub_key);
+        self.0.insert(address, update);
     }
 
-    pub(crate) fn push_update(&mut self, update: tendermint::validator::Update) {
-        self.0.insert(update.pub_key.to_hex(), update);
+    pub(crate) fn remove(&mut self, address: &account::Id) {
+        self.0.remove(address);
     }
 
     /// Apply updates to the validator set.
     ///
     /// If the power of a validator is set to 0, remove it from the set.
     /// Otherwise, update the validator's power.
     pub(crate) fn apply_updates(&mut self, validator_updates: ValidatorSet) {
-        for (pub_key, update) in validator_updates.0 {
+        for (address, update) in validator_updates.0 {
             match update.power.value() {
-                0 => self.0.remove(&pub_key),
-                _ => self.0.insert(pub_key, update),
+                0 => self.0.remove(&address),
+                _ => self.0.insert(address, update),
```

Comment:
Does it make sense to use the `remove` and `push_update` methods you defined above?

Reply:
I think this is okay since `push_update` recalculates the address which isn't needed